File: 575. Distribute Candies/575.go
package leetcode

func distributeCandies(candies []int) int {
	myDict := make(map[int]int, 0)
	for _, v := range candies{
		myDict[v]++
	}
	if len(myDict)<len(candies)/2{
		return len(myDict)
	}
	return len(candies)/2
}

File: 575. Distribute Candies/_575.py
class Solution:
    def distributeCandies(self, candies) -> int:
        myDict = {}
        for i in candies:
            myDict[i] = 1

        return min(len(myDict), len(candies)//2)

    # 再次强调去重想set
    def distributeCandies1(self, candies):
        return min(len(candies)//2, len(set(candies)))